cfgov/unprocessed/js/modules/util/format.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Turns a string into a number.
+ * Assumes each number in the string should be preserved (unlike parseInt)
+ * Assumes the first instance of a decimal point is the intended one.
+ * @param  {string} numberString - A string representing a number.
+ * @returns {number} The assumed numeric value of numberString
+ */
+function convertStringToNumber(numberString) {
+  if (typeof numberString === 'number') {
+    return numberString;
+  } else if (typeof numberString !== 'string') {
+    return 0;
+  }
+
+  let signMaker = 1;
+  const minusPosition = numberString.indexOf(numberString.match('-'));
+  const digitPosition = numberString.indexOf(numberString.match(/\d/));
+  const dashPosition = numberString.indexOf(numberString.match('-'));
+
+  // If a '-' appears before the first digit, we assume numberString is negative
+  if (dashPosition !== -1 && minusPosition < digitPosition) {
+    signMaker = -1;
+  }
+
+  // Strip non-numeric values, maintaining periods
+  numberString = numberString.replace(/[^0-9\.]+/g, '');
+
+  numberString = numberString.replace(/\./g, replaceCommas);
+
+  // Get number value of string, then multiply by signMaker and return
+  return Number(numberString) * signMaker;
+}
+
+/**
+ * This helper function places commas in the string. It's set up to
+ * be passed as a parameter to String.replace()
+ *
+ * @param {string} match - The matched substring.
+ * @param {number} offset - The numeric offset of the matched substring.
+ * @param {string} full - The full string to be matched against.
+ * @returns {string} new string to replace.
+ */
+function replaceCommas(match, offset, full) {
+  if (offset === full.indexOf('.')) {
+    return '.';
+  }
+  return '';
+}
```

Comment:
This function doesn't do what it says it does. This function ACTUALLY removes all but the first occurrence of a `.` when passed as the second argument to `String.replace`

I kinda think we should drop this functionality and use unformatUSD (as mentioned above). Right now we have very similar behavior in different parts of the site with different edge case handling. This seems suboptimal!

Reply:
Mmmm I had grabbed the longer comment from the [retirement copy of this](https://github.com/cfpb/consumerfinance.gov/blob/408cffe65975ddf9b608a091b0787de5756167e2/cfgov/unprocessed/apps/retirement/js/utils/handle-string-input.js#L27), but the [paying for college copy(https://github.com/cfpb/consumerfinance.gov/blob/e5979ffe9cec79dab4eef6bb6b78dea6e04502be/cfgov/unprocessed/apps/paying-for-college/js/disclosures/utils/handle-string-input.js#L33) would be more accurate.

---

I went the other way and replaced `unFormatUSD` with `convertStringToNumber`, as that was used far more than `unFormatUSD`. The functional changes are that in places `unFormatUSD` was used, the number `'123.45.67'`  will now end up as `'123.4567'` instead of `'123.45'`, and all non-string values passed in return zero instead of the original value passed in.

https://github.com/cfpb/consumerfinance.gov/pull/7400/commits/95b9434186ce557375bdd4f2374c7bbe19bdd928